### ISSUE TITLE
`tcp 16-20` some endpoints may be on the whitelist

### ISSUE THREAD

**Issue by SteZWRqU** [hyperion-cs/dpi-checkers#44]
В частности: 

```
US.CF-02: https://genshin.jmp.blue/characters/all#
US.CF-03: https://api.frankfurter.dev/v1/2000-01-01..2002-12-31
DE.HE-01: https://j.dejure.org/jcg/doctrine/doctrine_banner.webp
DE.HE-02: https://accesorioscelular.com/tienda/css/plugins.css
SE.OR-01: https://oracle.sfx.ovh/10M.bin
PL.AKM-01: https://media-assets.stryker.com/is/image/stryker/gateway_1?$max_width_1410$
US.CNST-01: https://cdn.xuansiwei.com/common/lib/font-awesome/4.7.0/fontawesome-webfont.woff2?v=4.7.0
```

Тестировал с помощью `curl`:

```
curl --output /dev/null --silent --show-error --max-time 5 --user-agent "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/145.0.0.0 Safari/537.36" --http1.1 --tlsv1.2 --tls-max 1.3 "https://$2"
```

Без использования `zapret`:

```
US.CF-01: Not detected
US.CF-02: Not detected
US.CF-03: Not detected
US.CF-04: curl: (28) Operation timed out after 5010 milliseconds with 18533 bytes received
US.DO-01: curl: (28) Operation timed out after 5007 milliseconds with 16196 out of 817776 bytes received
US.DO-02: curl: (28) Operation timed out after 5015 milliseconds with 16197 out of 129739 bytes received
US.DO-03: curl: (28) Operation timed out after 5013 milliseconds with 0 bytes received
DE.HE-01: Not detected
DE.HE-02: Not detected
FI.HE-01: curl: (28) Operation timed out after 5012 milliseconds with 16092 out of 1048576 bytes received
FI.HE-02: curl: (28) Operation timed out after 5011 milliseconds with 19196 out of 150124 bytes received
FI.HE-03: curl: (28) Operation timed out after 5011 milliseconds with 0 bytes received
FI.HE-04: curl: (28) Operation timed out after 5009 milliseconds with 16092 out of 1048576 bytes received
FR.OVH-01: curl: (28) Operation timed out after 5013 milliseconds with 0 bytes received
FR.OVH-02: curl: (28) Operation timed out after 5002 milliseconds with 7745 out of 676973 bytes received
SE.OR-01: Not detected
DE.AWS-01: curl: (28) Operation timed out after 5004 milliseconds with 16384 out of 76084 bytes received
US.AWS-01: curl: (28) Operation timed out after 5016 milliseconds with 15993 out of 685407 bytes received
US.GC-01: Not detected
US.FST-01: Not detected
CA.FST-01: Not detected
US.AKM-01: curl: (28) Operation timed out after 5016 milliseconds with 16384 out of 165202 bytes received
PL.AKM-01: Not detected
US.CDN77-01: curl: (28) Operation timed out after 5008 milliseconds with 16366 out of 65592 bytes received
FR.CNTB-01: curl: (28) Operation timed out after 5007 milliseconds with 23771 bytes received
NL.SW-01: curl: (28) Operation timed out after 5005 milliseconds with 0 bytes received
US.CNST-01: Not detected
```

Используя `zapret` с фейком `tls client hello`  с рандомным `sni` (чтобы быть уверенным, что его нет ни в белых, ни в черных списках), например: `fclmqhfwrk.ru`:

```
--filter-tcp=443 --dpi-desync=fake --dpi-desync-fake-tls-mod=rnd,dupsid,sni=fclmqhfwrk.ru --dpi-desync-fooling=ts
```

Результат:

```
US.CF-01: Not detected
US.CF-02: curl: (28) Operation timed out after 5012 milliseconds with 1590 out of 639522 bytes received
US.CF-03: curl: (28) Operation timed out after 5010 milliseconds with 23273 out of 290660 bytes received
US.CF-04: curl: (28) Operation timed out after 5011 milliseconds with 2732 bytes received
US.DO-01: curl: (28) Operation timed out after 5010 milliseconds with 0 bytes received
US.DO-02: curl: (28) Operation timed out after 5007 milliseconds with 0 bytes received
US.DO-03: curl: (28) Operation timed out after 5012 milliseconds with 0 bytes received
DE.HE-01: curl: (28) Operation timed out after 5009 milliseconds with 0 bytes received
DE.HE-02: curl: (28) Operation timed out after 5011 milliseconds with 17943 out of 598883 bytes received
FI.HE-01: curl: (28) Operation timed out after 5003 milliseconds with 0 bytes received
FI.HE-02: curl: (28) Operation timed out after 5016 milliseconds with 23813 out of 150124 bytes received
FI.HE-03: curl: (28) Operation timed out after 5001 milliseconds with 16092 out of 1048576 bytes received
FI.HE-04: curl: (28) Operation timed out after 5012 milliseconds with 16092 out of 1048576 bytes received
FR.OVH-01: curl: (28) Operation timed out after 5011 milliseconds with 0 bytes received
FR.OVH-02: curl: (28) Operation timed out after 5017 milliseconds with 0 out of 676973 bytes received
SE.OR-01: curl: (28) Operation timed out after 5008 milliseconds with 0 bytes received
DE.AWS-01: curl: (28) Operation timed out after 5002 milliseconds with 0 out of 76084 bytes received
US.AWS-01: curl: (28) Operation timed out after 5011 milliseconds with 16384 out of 685407 bytes received
US.GC-01: Not detected
US.FST-01: Not detected
CA.FST-01: Not detected
US.AKM-01: curl: (28) Operation timed out after 5006 milliseconds with 15968 out of 165202 bytes received
PL.AKM-01: curl: (28) Operation timed out after 5012 milliseconds with 16384 out of 96498 bytes received
US.CDN77-01: curl: (28) Operation timed out after 5003 milliseconds with 0 bytes received
FR.CNTB-01: curl: (28) Operation timed out after 5004 milliseconds with 7387 bytes received
NL.SW-01: curl: (28) Operation timed out after 5014 milliseconds with 0 bytes received
US.CNST-01: curl: (28) Operation timed out after 5013 milliseconds with 0 bytes received
```

Также с `zapret`, но с `sni` из белого списка:

```
US.CF-01: Not detected
US.CF-02: Not detected
US.CF-03: Not detected
US.CF-04: Not detected
US.DO-01: Not detected
US.DO-02: Not detected
US.DO-03: Not detected
DE.HE-01: Not detected
DE.HE-02: Not detected
FI.HE-01: Not detected
FI.HE-02: Not detected
FI.HE-03: Not detected
FI.HE-04: Not detected
FR.OVH-01: Not detected
FR.OVH-02: Not detected
SE.OR-01: Not detected
DE.AWS-01: Not detected
US.AWS-01: Not detected
US.GC-01: Not detected
US.FST-01: Not detected
CA.FST-01: Not detected
US.AKM-01: Not detected
PL.AKM-01: Not detected
US.CDN77-01: Not detected
FR.CNTB-01: Not detected
NL.SW-01: Not detected
US.CNST-01: Not detected
```

**Comment by hyperion-cs** [comment]
@SteZWRqU, сensor works in such a way that it does not restrict every connection. Try running the tests about 5-10 times (be sure to use different connections, no keep-alived or connection pools) — will the results be the same?

It is also worth noting that your provider may have these endpoints on their whitelist (by domain name and/or subnet), while others may not. Therefore, we only replace them if we understand that this is a common issue.

For example, `DE.HE-02` is marked as `tcp 16-20 `detected in my case, and `US.CF-02` and `US.CF-03` fail due to connection timeout.

**Comment by SteZWRqU** [comment]
> It is also worth noting that your provider may have these endpoints on their whitelist (by domain name and/or subnet), while others may not. Therefore, we only replace them if we understand that this is a common issue.

Мой провайдер Ростелеком, Санкт-Петербург, так что, я думаю, это затрагивает многих

> сensor works in such a way that it does not restrict every connection. Try running the tests about 5-10 times (be sure to use different connections, no keep-alived or connection pools) — will the results be the same?

Как я выше писал, я использую не ваш сервис в браузере, а `curl`, очевидно, что каждый раз он использует новое соединение

От запуска к запуску может быть как:
```
curl: (28) Operation timed out after 5016 milliseconds with 0 bytes received
```
Так и:
```
curl: (28) Operation timed out after 5014 milliseconds with 23496 out of 639522 bytes received
```

Поясню мою логику:

Если к подсети не применяется блокировка `tcp 16-20`, то в этой подсети должен работать любой `sni`, в том числе и мой рандомный `fclmqhfwrk.ru` 

Если же в подсети работает, например, `genshin.jmp.blue`, но не работает `fclmqhfwrk.ru` значит `genshin.jmp.blue` в белом списке


Еще стоит отметить, что в `curl` я явно указываю `--http1.1 --tlsv1.2 --tls-max 1.3` чтобы быть уверенным, что используется именно `tls (tcp)`, в то время как браузер может преключится на `quic (udp)` . Это может повлиять на результат, например, у моего провайдера разные белые списки для `tls (tcp)` и для `quic (udp)`

**Comment by hyperion-cs** [comment]
@SteZWRqU, got it. Overall, your logic is correct, but not quite complete. A censor in the `tcp 16-20` restriction method has, in general terms:
1. Blacklist by subnets that triggers "sometimes";
2. Blacklist by subnets that triggers "always";
3. Blacklist by domains that triggers "sometimes";
4. Blacklist by domains that triggers "always";
5. Whitelist by subnets that triggers "sometimes" (_inside the blacklist by domains_);
6. Whitelist by subnets that triggers "always" (_inside the blacklist by domains_);
7. Whitelist by domains that triggers "sometimes" (_inside the blacklist by subnets_);
8. Whitelist by domains that triggers "always" (_inside the blacklist by subnets_).

In addition, _whitelist > blacklist_. Perhaps those things that work "sometimes" can be explained by the imperfection of DPI (in both directions — allow and disallow), but in any case, they are observed and depend on domains/subnets. Thus, it is quite difficult to detect them (especially those that are triggered only sometimes) but we are trying.

---
We will conduct an additional analysis of the endpoints you specified. If it turns out that they are indeed mostly on the whitelist, we will try to replace them.

**Comment by TestPolygon** [comment]
I confirm only these two hosts:
```
Cloudflare US.CF-02: https://genshin.jmp.blue/characters/all
Cloudflare US.CF-03: https://api.frankfurter.dev/v1/2000-01-01..2002-12-31
```
They are 100 % whitelisted. They do not work with the enabled host corrupting like whitelisted `sourceforge.net`, `vite.dev`.

Also, it looks, the block is up to 28 KB (?), not 20 KB.
```bash
curl -w '%{size_download}\n' -o /dev/null https://genshin.jmp.blue/characters/all
```
```bash
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  4  624k    4 27369    0     0   1107      0  0:09:37  0:00:24  0:09:13     0
```
```bash
curl -w '%{size_download}\n' -o /dev/null https://api.frankfurter.dev/v1/2000-01-01..2002-12-31
```
```bash
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  9  283k    9 27380    0     0   1887      0  0:02:34  0:00:14  0:02:20     0
```

**Comment by hyperion-cs** [comment]
@TestPolygon 

>They are 100 % whitelisted

Well, my provider has a connection timeout for these endpoints. It is worth noting that resources passed through CF will have different outgoing IP addresses depending on geolocation and other factors (some may be restricted, some may not).
For example, in my case:
- _US.CF-02_ => _172.67.190.137_
- _US.CF-03_ => _104.21.26.111_

\* both by [AS13335](https://bgp.tools/as/13335)

>Also, it looks, the block is up to 28 KB (?), not 20 KB.

Very interesting. I still have around 16KB. We'll keep an eye on it and consider increasing the threshold to 32KB in the checker (currently around 24KB).

**Comment by SteZWRqU** [comment]
Похоже Вы меня так и не поняли, я говорю о том, что указанные мной `endpoints` находятся в белом списке для `tls (tcp)` у моего провайдера:

```
US.CF-02: https://genshin.jmp.blue/characters/all#
US.CF-03: https://api.frankfurter.dev/v1/2000-01-01..2002-12-31
DE.HE-01: https://j.dejure.org/jcg/doctrine/doctrine_banner.webp
DE.HE-02: https://accesorioscelular.com/tienda/css/plugins.css
SE.OR-01: https://oracle.sfx.ovh/10M.bin
PL.AKM-01: https://media-assets.stryker.com/is/image/stryker/gateway_1?$max_width_1410$
US.CNST-01: https://cdn.xuansiwei.com/common/lib/font-awesome/4.7.0/fontawesome-webfont.woff2?v=4.7.0
```

Еще раз подробно объясню как я это определил на примере `https://genshin.jmp.blue/characters/all#`:

Я делаю запрос с помощью `curl` к `https://genshin.jmp.blue/characters/all#`:
```
curl --output /dev/null --silent --show-error --max-time 5 --user-agent "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/145.0.0.0 Safari/537.36" --http1.1 --tlsv1.2 --tls-max 1.2 "https://genshin.jmp.blue/characters/all#"
```
Так как я не применяю `zapret`, DPI видит оригинальный `TLS ClientHello` с `sni=genshin.jmp.blue` и пропускает этот запрос, `curl` получает полный ответ от сервера (сколько бы раз я не делал этот запрос - я всегда получаю полный ответ)

Дальше я запускаю `zapret`:

```
--filter-tcp=443 --dpi-desync=fake --dpi-desync-fake-tls-mod=rnd,dupsid,sni=fclmqhfwrk.ru --dpi-desync-fooling=ts
```

`zapret` в данном случае при каждом `tls` запросе посылает фейковый `TLS ClientHello` с `sni=fclmqhfwrk.ru`, чтобы обмануть DPI, домен `fclmqhfwrk.ru` используемый в `sni` составлен из случайных букв, чтобы гарантировать, что этого домена нет ни в белых, ни в черных списках

Далее с запущенным `zapret` я снова делаю запрос `curl` к `https://genshin.jmp.blue/characters/all#`. DPI видит фейковый `TLS ClientHello` с `sni=fclmqhfwrk.ru` и применяет блокировку `tcp 16-20`, а я получаю не полный ответ от сервера (сколько бы раз я не делал этот запрос - я всегда получаю не полный ответ)

Далее я перезапускаю `zapret`, чтобы он посылал фейковый `TLS ClientHello` с `sni` из белого списка (раньше это был `max.ru`, сейчас я использую другой):

```
--filter-tcp=443 --dpi-desync=fake --dpi-desync-fake-tls-mod=rnd,dupsid,sni=max.ru --dpi-desync-fooling=ts
```

Снова посылаю запрос `curl` к `https://genshin.jmp.blue/characters/all#`. DPI видит фейковый `TLS ClientHello` с `sni=max.ru` и пропускает этот запрос, `curl` получает полный ответ от сервера (сколько бы раз я не делал этот запрос - я всегда получаю полный ответ)

Из этого делаю вывод, что применение блокировки `tcp 16-20` происходит не потому что я использую `zapret` и DPI это обнаруживает, а потому что `fclmqhfwrk.ru` не находится в белом списке. И из этого же следует, что `genshin.jmp.blue` находится в белом списке

**Comment by TestPolygon** [comment]
After a retest in a new incognito tab with the enabled host/SNI corrupting I confirm these hosts are also whitelisted:
```
Hetzner DE.HE-01: https://j.dejure.org/jcg/doctrine/doctrine_banner.webp
Hetzner DE.HE-02: https://accesorioscelular.com/tienda/css/plugins.css
Oracle  SE.OR-01: https://oracle.sfx.ovh/10M.bin
Akamai PL.AKM-01: https://media-assets.stryker.com/is/image/stryker/gateway_1?$max_width_1410$
Vultr US.VLTR-01: https://cdn.xuansiwei.com/common/lib/font-awesome/4.7.0/fontawesome-webfont.woff2?v=4.7.0
```

<img width="764" height="44" alt="Image" src="https://github.com/user-attachments/assets/12de43ac-b6fc-4e48-93f2-178816a1a700" />

<img width="418" height="286" alt="Image" src="https://github.com/user-attachments/assets/22ce4a84-707b-4daf-8a81-554fb4acfda2" />

<img width="418" height="278" alt="Image" src="https://github.com/user-attachments/assets/2661d4f9-04b3-42f9-bebf-ff6ddb7d9d55" />

**Comment by hyperion-cs** [comment]
@TestPolygon,
> with the enabled host corrupting

What is this? It would seem that any "corrupting" during testing for inclusion in the whitelist should be disabled...
I also wouldn't put too much trust in standard downloads via a browser, as this can have side effects in the form of keep-alived, connection pools to hosts, etc.

**Comment by TestPolygon** [comment]
> What is this?

`-7`

> via a browser ...

It's why I always use a fresh (reopened) incognito browser _window_ (not just a new tab) for tests.

**Comment by hyperion-cs** [comment]
@SteZWRqU, please test these endpoints with your provider:

| \# | Provider | Endpoint |
|:-:|:-:|-|
| 1.1 | Oracle | https://global-seres.com.sg/wp-content/uploads/2024/02/SVG00732-scaled.jpg |
| 1.2 | Oracle | https://mesajil.com/ |
| 1.3 | Oracle | https://zojirushi.sg/wp-content/uploads/2025/12/SM-VH60-95_00.jpg |
| 2.1 | Contabo | https://www.cateringexner.cz/font/ebrima/ebrima.woff2 |
| 2.2 | Contabo | https://findair.net/wp-content/uploads/2025/07/online-booking-2.jpeg |
| 2.3 | Contabo | https://app.panel-it.cz/node_modules/weathericons/font/weathericons-regular-webfont.woff |
| 3.1 | Vultr | https://player.sakaserver.com/web/modules/fonts/material-icons/mi_2023_11.woff2 |
| 3.2 | Vultr | https://us.rudder.qntmnet.com/QN-CDN/images/qn_bg_.jpg |
| 3.3 | Vultr | https://tressonuno.com/utilidades/2/1.png |
| 4.1 | Hetzner | https://qrimp.net/js/jquery-1.6.4.min.js |
| 4.2 | Hetzner | https://apiwhatsapp-1000.zapipro.com/libs/bootstrap/dist/css/bootstrap.min.css |
| 4.3 | Hetzner | https://walk.skaffen.org/cover/andromeda.png |
| 4.4 | Hetzner | https://mapeak.com/content/hero-mountain.jpg |
| 4.5 | Hetzner | https://higuma-craftbeer.fr/wp-content/themes/pub-store/fontsawesome/webfonts/fa-solid-900.woff2 |
| 4.6 | Hetzner | https://www.industrialport.net/wp-content/uploads/custom-fonts/2022/10/Lato-Bold.ttf |
| 4.7 | Hetzner | https://aryanor.com/wp-content/uploads/2024/02/S2.webp |
| 5.1 | Akamai | https://www.mobilelegends.com/assets/utils-790bf987.js |
| 5.2 | Akamai | https://akmweb.youngjoygame.com/web/gms/image/709fa57eb71b64f59861e89572dcae76.jpg |
| 5.3 | Akamai | https://www.miele.com/media/common/fonts/MieleElements/MieleElements-5Regular.woff2 |
| 6.1 | CF | https://docs.maptiler.com/assets/fonts/Open_Sans/OpenSans-Italic-VariableFont_wdth,wght.ttf |
| 6.2 | CF | https://wsrv.nl/assets/la-solid-900.TjMEgv3Q.woff2 |
| 6.3 | CF | https://api.codetabs.com/v1/random/integer?min=1111111111&max=9999999999&times=10000 |
| 6.4 | CF | https://esm.sh/gh/esm-dev/esm.sh@e7447dea04/server/embed/assets/sceenshot-deno-types.png |
| 6.5 | CF | https://unpkg.com/vue@3.5.28/dist/vue.esm-browser.prod.js |
| 6.6 | CF | https://api.allorigins.win/raw?url=https%3A%2F%2Fwww.wikipedia.org%2F |

\* Other participants in this issue can also share their tests of this.

**Comment by Shararamosh** [comment]
Лично у меня из всех перечисленных в первом комментарии адресов стабильно, вне зависимости от того, есть запрет или нет, работает только SE.OR-01. OVH стабильно перестают работать (при этом не из-за торрента, как все пишут) каждые 10 минут, Hetzner, все Cloudflare (кроме US.CF-01) и все Akamai работают исключительно с фейками. При этом PL.AKM-01 бо́льшую часть времени не работает даже с фейками.
Проверял на домашнем Билайне, мобильном Билайне, Дом.ру и Orange Business.

**Comment by TestPolygon** [comment]
After quick testes the links above look valid (not whitelisted).

---

The related note: Before doing the experiments try to toggle (disable and then enable) the network adapter in Windows settings to ensure that all network connections (and drivers) were reset.

<img width="801" height="382" alt="https://geekrewind.com/wp-content/uploads/2022/02/disable-network-adapter-windows-11-control-panel.webp" src="https://github.com/user-attachments/assets/48fade45-ae30-41ea-bf62-3e293134de9c" />

**Comment by SteZWRqU** [comment]
Для начала проверил через VPN (Cloudflare WARP):

```
1.1: Not detected
1.2: curl: (28) Connection timed out after 10011 milliseconds
1.3: Not detected
2.1: Not detected
2.2: Not detected
2.3: curl: (28) Connection timed out after 10005 milliseconds
3.1: Not detected
3.2: Not detected
3.3: Not detected
4.1: Not detected
4.2: Not detected
4.3: Not detected
4.4: curl: (28) Connection timed out after 10016 milliseconds
4.5: Not detected
4.6: Not detected
4.7: Not detected
5.1: Not detected
5.2: Not detected
5.3: curl: (28) Connection timed out after 10005 milliseconds
6.1: Not detected
6.2: Not detected
6.3: Not detected
6.4: Not detected
6.5: Not detected
6.6: curl: (28) Operation timed out after 10014 milliseconds with 0 bytes received
```

Следующие `endpoints` или работают не стабильно, или вовсе не работают:
```
1.2: Не работает
2.3: Не работает
4.4: Не работает
5.3: Не работает
6.6: Очень часто возвращает 500 Internal Server Error
```

Поэтому, думаю, их стоит сразу исключить.
<details>
<summary>Но тем не менее, я все же проверил их на своем провайдере:</summary>

Без `zapret`:
```
1.2: curl: (28) Operation timed out after 10011 milliseconds with 0 bytes received
2.3: curl: (28) Operation timed out after 10017 milliseconds with 0 bytes received
4.4: curl: (28) Operation timed out after 10011 milliseconds with 0 bytes received
5.3: curl: (28) Operation timed out after 10005 milliseconds with 0 bytes received
6.6: curl: (28) Operation timed out after 10015 milliseconds with 0 bytes received
```

`zapret` с рандомным `sni` (`craxvimfda.ru`):
```
1.2: curl: (28) Operation timed out after 10016 milliseconds with 0 bytes received
2.3: curl: (28) Operation timed out after 10010 milliseconds with 0 bytes received
4.4: curl: (28) Operation timed out after 10002 milliseconds with 0 bytes received
5.3: curl: (28) Operation timed out after 10015 milliseconds with 0 bytes received
6.6: curl: (28) Operation timed out after 10014 milliseconds with 0 bytes received
```

`zapret` с `sni` из белого списка:
```
1.2: curl: (28) Operation timed out after 10004 milliseconds with 0 bytes received
2.3: Not detected
4.4: Not detected
5.3: Not detected
6.6: curl: (28) Operation timed out after 10004 milliseconds with 0 bytes received
```
</details>


Эти я не могу проверить:
```
4.5: Его ip (144.76.8.3) заблокирован на моем провайдере
4.7: Не резолвится через мой DNS (Yandex), DNS 1.1.1.1 дает ip 144.76.9.138 - он заблокирован на моем провайдере
```

Оставшиеся:

Без `zapret`:
```
1.1: curl: (28) Operation timed out after 10001 milliseconds with 23184 out of 213263 bytes received
1.3: curl: (28) Operation timed out after 10010 milliseconds with 0 bytes received
2.1: curl: (28) Operation timed out after 10003 milliseconds with 0 bytes received
2.2: curl: (28) Operation timed out after 10011 milliseconds with 0 bytes received
3.1: curl: (28) Operation timed out after 10008 milliseconds with 0 bytes received
3.2: curl: (28) Operation timed out after 10004 milliseconds with 14980 out of 77930 bytes received
3.3: curl: (28) Operation timed out after 10014 milliseconds with 15963 out of 100880 bytes received
4.1: curl: (28) Operation timed out after 10014 milliseconds with 0 bytes received
4.2: curl: (28) Operation timed out after 10001 milliseconds with 0 bytes received
4.3: curl: (28) Operation timed out after 10011 milliseconds with 15545 out of 1840252 bytes received
4.6: curl: (28) Operation timed out after 10011 milliseconds with 0 bytes received
5.1: Not detected
5.2: Not detected
6.1: Not detected
6.2: Not detected
6.3: curl: (28) Operation timed out after 10009 milliseconds with 25402 bytes received
6.4: Not detected
6.5: Not detected
```

`zapret` с рандомным `sni` (`craxvimfda.ru`):
```
1.1: curl: (28) Operation timed out after 10006 milliseconds with 0 bytes received
1.3: curl: (28) Operation timed out after 10009 milliseconds with 0 bytes received
2.1: curl: (28) Operation timed out after 10009 milliseconds with 0 bytes received
2.2: curl: (28) Operation timed out after 10005 milliseconds with 0 bytes received
3.1: curl: (28) Operation timed out after 10014 milliseconds with 0 bytes received
3.2: curl: (28) Operation timed out after 10001 milliseconds with 0 bytes received
3.3: curl: (28) Operation timed out after 10014 milliseconds with 0 bytes received
4.1: curl: (28) Operation timed out after 10006 milliseconds with 0 bytes received
4.2: curl: (28) Operation timed out after 10007 milliseconds with 0 bytes received
4.3: curl: (28) Operation timed out after 10003 milliseconds with 0 bytes received
4.6: curl: (28) Operation timed out after 10006 milliseconds with 0 bytes received
5.1: curl: (28) Operation timed out after 10001 milliseconds with 0 bytes received
5.2: curl: (28) Operation timed out after 10002 milliseconds with 0 bytes received
6.1: curl: (28) Operation timed out after 10013 milliseconds with 2732 bytes received
6.2: curl: (28) Operation timed out after 10017 milliseconds with 2738 out of 96752 bytes received
6.3: curl: (28) Operation timed out after 10003 milliseconds with 3497 bytes received
6.4: Not detected
6.5: curl: (28) Operation timed out after 10003 milliseconds with 23758 out of 166945 bytes received
```

`zapret` с `sni` из белого списка:
```
1.1: Not detected
1.3: Not detected
2.1: Not detected
2.2: Not detected
3.1: Not detected
3.2: Not detected
3.3: Not detected
4.1: Not detected
4.2: Not detected
4.3: Not detected
4.6: Not detected
5.1: Not detected
5.2: Not detected
6.1: Not detected
6.2: Not detected
6.3: Not detected
6.4: Not detected
6.5: Not detected
```

Из этого делаю вывод, что `5.1`, `5.2`, `6.1`, `6.2` и `6.5` в белом списке

**Comment by SteZWRqU** [comment]
Так же в url `6.3` Вы случайно добавили лишнюю букву `F` в конце. В своих тестах я ее убрал

**Comment by hyperion-cs** [comment]
@SteZWRqU thank you for your report. We will take this into account in the next update.

**Comment by hyperion-cs** [comment]
@TestPolygon by the way, an interesting question about connections. We need to analyze how we handle them (I can't remember right now). Note that for Chrome, you can reset connections in _chrome://net-internals/#sockets_

**Comment by TestPolygon** [comment]
I did not tested it before carefully. 
After multiple retests with `-7` and without it, I confirm only `# 5.1 Akamai` and `# 5.2 Akamai` are whitelisted, however, sometimes they may have a notable delay before the request is completed.

**Comment by TestPolygon** [comment]
~Wait.~
```
US.CF-02: https://genshin.jmp.blue/characters/all
US.CF-03: https://api.frankfurter.dev/v1/2000-01-01..2002-12-31
```
~are not whitelisted for me anymore.~

~And https://sourceforge.net/ too.~

UPD: It's was a temporal entire CF blocking.

**Comment by TestPolygon** [comment]
BTW, Cloudflare has increased CWND (Congestion Window) — 30 MSS (~43 KB), not 10 MSS (~14 KB).
It's, possibly, why the count of downloaded KBs from Cloudflare may be more than standard 16-20 KB.

**Comment by hyperion-cs** [comment]
TODO:
- [x] US.CF-02
- [x] US.CF-03
- [x] DE.HE-01
- [x] DE.HE-02
- [x] SE.OR-01
- [x] PL.AKM-01
- [x] US.CNST-01
- [x] FR.CNTB-01
- [x] FR.OVH-02

**Comment by hyperion-cs** [comment]
Partially fixed in bc92ede016cb6e94d2c480400eaf0cabd8f87feb.

**Comment by hyperion-cs** [comment]
@SteZWRqU, the most problematic endpoints are always where the CDN is (the result is very unstable due to the client's location, balancing, sharing their networks with other CDN, etc.) — currently, these are CF and Akamai. However, please check these endpoints (they all work for me):

| \# | Provider | Endpoint |
|:-:|:-:|-|
| 1.1 | CF | https://localseptictx.com/wp-content/uploads/2019/12/pexels-photo-2449603.jpg |
| 1.2 | CF | https://cdn.prod.website-files.com/67ff66f19400bec8576163e4/684ac37e311547a6edb9c3dc_5f79e3049193d892595c0a4bd01b13ad_Knock.webp |
| 1.3 | CF | https://www.xuperblimited.com/cdn/shop/files/202304180829452.jpg |
| 1.4 | CF | https://cdn.xiaoman.cn/mkt_web/widget/0.1.49/static/js/vendor.1544171b.js |
| 1.5 | CF | https://pddup.pics/ |
| 1.6 | CF | https://prod.embed.conversations.dealerinspire.com/app.js |
| 1.7 | CF | https://aegis.audioeye.com/assets/index.js |
| 1.8 | CF | https://sierra.chat/-/cdn/wPz02HKDMp9OTBfx4ogVYU0FjIU.js |
| 2.1 | Akamai | https://images.dealer.com/dbcreative/szottm59toyotascion/D-10733/2026-02-szottm59toyotascion-D-10733-bZ-SL.jpg |
| 2.2 | Akamai | https://shanaford.herbalife.com/dmassets/ds-admin/global-reusable-assets/images/ri-blackberrry-boost-spritzer.jpg |
| 2.3 | Akamai | https://capi.myleasestar.com/v2/dimg/151961177/1920x1400/151961177.jpg |
| 2.4 | Akamai | https://media.miele.com/images/2000015/200001503/20000150334.png |
| 2.5 | Akamai | https://www.fbgcdn.com/websites/fonts/pt-serif-v17-latin-ext_latin_cyrillic-ext_cyrillic-regular.woff2 |
| 2.6 | Akamai | https://credify.tnn.in/loginWidget.1.0.1.js |
| 2.7 | Akamai | https://www.timesnownews.com/nimg/assets/livetv-logo/times-now-navbharat.svg |
| 2.8 | Akamai | https://staticw2.yotpo.com/AoYixD1Gp8g9O6lWsWgIYhdAHFmCS4hgtfgd84X4/widget.js |

**Comment by SteZWRqU** [comment]
Нет возможности проверить:

**1.4**: `zapret` его ломает, вероятно сервер не поддерживает `TCP Timestamps`
**1.6**: И на моем провайдере и через VPN (Cloudflare WARP) возвращает статус `403`, и соответственно, слишком короткий ответ

Без `zapret`:
```
1.1: Not detected
1.2: Not detected
1.3: Not detected
1.4: Not detected
1.5: curl: (28) Operation timed out after 15015 milliseconds with 22276 bytes received
1.6: Not detected
1.7: curl: (28) Operation timed out after 15009 milliseconds with 8185 bytes received
1.8: Not detected
2.1: curl: (28) Operation timed out after 15007 milliseconds with 15914 out of 65536 bytes received
2.2: curl: (28) Operation timed out after 15003 milliseconds with 0 bytes received
2.3: curl: (28) Operation timed out after 15014 milliseconds with 0 bytes received
2.4: curl: (28) Operation timed out after 15012 milliseconds with 15949 out of 65536 bytes received
2.5: curl: (28) Operation timed out after 15007 milliseconds with 24576 bytes received
2.6: curl: (28) Operation timed out after 15006 milliseconds with 15914 bytes received
2.7: curl: (28) Operation timed out after 15001 milliseconds with 16384 bytes received
2.8: curl: (28) Operation timed out after 15004 milliseconds with 15702 bytes received
```

`zapret` с рандомным `sni` (`craxvimfda.ru`):
```
1.1: Not detected
1.2: curl: (28) Operation timed out after 15015 milliseconds with 2017 out of 65536 bytes received
1.3: Not detected
1.4: curl: (28) Connection timed out after 15006 milliseconds
1.5: curl: (28) Operation timed out after 15012 milliseconds with 1733 bytes received
1.6: Not detected
1.7: curl: (28) Operation timed out after 15017 milliseconds with 1363 bytes received
1.8: Not detected
2.1: curl: (28) Operation timed out after 15015 milliseconds with 15914 out of 65536 bytes received
2.2: curl: (28) Operation timed out after 15009 milliseconds with 0 bytes received
2.3: curl: (28) Operation timed out after 15012 milliseconds with 0 bytes received
2.4: curl: (28) Operation timed out after 15002 milliseconds with 15949 out of 65536 bytes received
2.5: curl: (28) Operation timed out after 15014 milliseconds with 0 bytes received
2.6: curl: (28) Operation timed out after 15016 milliseconds with 0 bytes received
2.7: curl: (28) Operation timed out after 15012 milliseconds with 0 bytes received
2.8: curl: (28) Operation timed out after 15009 milliseconds with 0 bytes received
```

`zapret` с `sni` из белого списка:
```
1.1: Not detected
1.2: Not detected
1.3: Not detected
1.4: curl: (28) Connection timed out after 15016 milliseconds
1.5: Not detected
1.6: Not detected
1.7: Not detected
1.8: Not detected
2.1: Not detected
2.2: Not detected
2.3: Not detected
2.4: Not detected
2.5: Not detected
2.6: Not detected
2.7: Not detected
2.8: Not detected
```

Так же во всех тестах использовались одинаковые ip:
```
1.1: 141.193.213.10
1.2: 104.18.161.117
1.3: 23.227.38.74
1.4: 47.246.2.212
1.5: 104.21.73.122
1.6: 104.18.32.16
1.7: 104.18.29.155
1.8: 8.47.69.0
2.1: 184.86.251.14
2.2: 23.210.17.188
2.3: 23.213.161.201
2.4: 23.201.43.72
2.5: 184.86.251.11
2.6: 2.22.31.177
2.7: 23.55.163.157
2.8: 2.17.241.219
```

Из этого делаю вывод, что `1.2` в белом списке.
`1.1`, `1.3`, `1.8` находятся на ip к которым мой провайдер не применяет блокировку `tcp 16-20`, соответственно невозможно проверить, есть ли они в белом списке

**Comment by SteZWRqU** [comment]
На мой взгляд, лучшие кандидаты для Cloudflare: `1.5` и `1.7`, для Akamai: `2.1` и `2.4`

**Comment by hyperion-cs** [comment]
@SteZWRqU thanks for the report. Closed in a29df6a7cfbf3e4253fe1139108c075c11b30adf.

**Comment by TestPolygon** [comment]
All Contabo (🇫🇷) (`cateringexner.cz`, `findair.net`) and DigitalOcean (🇺🇸) (`carishealthcare.com`, `bohnlawllc.com`, `ecomstal.com`) were whitelisted, BTW.

UPD (after 6 hours):
Seems, these too now: Cloudflare (🇨🇦) (`bigcartel.com`, `audioeye.com`), Hetzner (🇩🇪) (`zapipro.com`, `industrialport.net`), Oracle (🇨🇴, 🇸🇬) (`plataforma.trackerintl.com`, `global-seres.com.sg`), OVH (🇫🇷) (`testing.symarobot.com`, `filmoteka.net.pl`)

UPD (after 10 hours):
Also, Vultr (🇩🇪) `static-cdn.play.date` and Melbicom (🇺🇸) `twin.mentat.su`.

**Comment by TestPolygon** [comment]
I have updated the message above 2 times: 15 domains are total for now.

**Comment by hyperion-cs** [comment]
It seems we have been able to solve this problem with a completely new approach. Stay tuned for the release soon.

**Comment by hyperion-cs** [comment]
A new checking method has been implemented, so errors and replacement of some endpoints are possible in the near future.
Closed in https://github.com/hyperion-cs/dpi-checkers/pull/55.

@SteZWRqU, @TestPolygon please note the new utility for **internal** use at [/utils/tcp1620_prober.py ](https://github.com/hyperion-cs/dpi-checkers/blob/main/utils/tcp1620_prober.py)— it is a comprehensive and powerful prober for a single host (domain name or ip). Among other things, you no longer need to use _zapret_ to "fool" the censor with sni to determine if it is whitelisted or not. For details, see the PR above.

<img width="944" height="813" alt="Image" src="https://github.com/user-attachments/assets/d0b75d69-0cdb-44e2-b0d7-51bb229cb914" />

and:
<img width="688" height="155" alt="Image" src="https://github.com/user-attachments/assets/b97d8e93-32df-4115-ac2f-9961fcb28c0a" />

**Comment by TestPolygon** [comment]
I would like to see more CF hosts from the blocked IP range, like `4chan.org`, `justice.gov`, `freedom.gov` (?, it looks WhiteListed, but with ECH),  `etcher.balena.io`,  `zfsbootmenu.org`, `stamps.am` _(also with ECH)_, `kgcorporation.jp`, `hono.dev`.  _(It's not a problem to find a CF host. There are _a lot_ of CF sites.)_

Also, FranTech / BuyVM _(as it was requested in the `Runnin4ik/dpi-detector`'s topic)_: `buyvm.net`, `my.frantech.ca`, `dmvideo.download`, `quetzalcoatl-relays.org`, `thenetnaija.ng`, `ceps.uni.edu.pe`, `mt-chambers.com`, `gazellegames.net`.

**Comment by hyperion-cs** [comment]
@TestPolygon I added two of each in 2f524444d8c0182464155d96912ab55302502367 (we'll keep the others for the future, in case the current ones get whitelisted).
Please create a new issue for similar requests, as the current one is already closed and quite large.
